tests/unit/test_fuzz.cpp

```diff
@@ -0,0 +1,161 @@
+{
+    return &g_TransmitRadioPacket;
+}
+
+void TestFuzz(uint32_t aSeconds)
+{
+    // Set the radio capabilities to disable any Mac related timer dependencies
+    g_testPlatRadioCaps = (otRadioCaps)(kRadioCapsAckTimeout | kRadioCapsTransmitRetries);
+
+    // Set the platform function pointers
+    g_TransmitRadioPacket.mPsdu = g_TransmitPsdu;
+    g_testPlatRadioReceive = testFuzzRadioReceive;
+    g_testPlatRadioTransmit = testFuzzRadioTransmit;
+    g_testPlatRadioGetTransmitBuffer = testFuzztRadioGetTransmitBuffer;
+
+    // Initialize our timing variables
+    uint32_t tStart = otPlatAlarmGetNow();
+    uint32_t tEnd = tStart + (aSeconds * 1000);
+
+    otInstance *aInstance;
+
+#ifdef OPENTHREAD_MULTIPLE_INSTANCE
+    uint64_t otInstanceBufferLength = 0;
+    uint8_t *otInstanceBuffer = NULL;
+
+    // Call to query the buffer size
+    (void)otInstanceInit(NULL, &otInstanceBufferLength);
+
+    // Call to allocate the buffer
+    otInstanceBuffer = (uint8_t *)malloc(otInstanceBufferLength);
+    VerifyOrQuit(otInstanceBuffer != NULL, "Failed to allocate otInstance");
+    memset(&otInstanceBuffer, 0, otInstanceBufferLength);
+
+    // Initialize Openthread with the buffer
+    aInstance = otInstanceInit(otInstanceBuffer, &otInstanceBufferLength);
+#else
+    aInstance = otInstanceInit();
+#endif
+
+    VerifyOrQuit(aInstance != NULL, "Failed to initialize otInstance");
+
+    // Start the Thread network
+    otSetPanId(aInstance, (otPanId)0xFACE);
+    otInterfaceUp(aInstance);
+    otThreadStart(aInstance);
+
+    uint32_t countRecv = 0;
+
+    while (otPlatAlarmGetNow() < tEnd)
+    {
+        otProcessQueuedTasklets(aInstance);
+
+        if (g_testPlatAlarmSet && otPlatAlarmGetNow() >= g_testPlatAlarmNext)
+        {
+            g_testPlatAlarmSet = false;
+            otPlatAlarmFired(aInstance);
+        }
+
+        if (g_fTransmit)
+        {
+            g_fTransmit = false;
+            otPlatRadioTransmitDone(aInstance, true, kThreadError_None);
+        }
+
+        if (g_RecvChannel != 0)
+        {
+            uint8_t fuzzRecvBuff[128];
+            RadioPacket fuzzPacket;
+
+            // Initialize the radio packet with a random length
+            memset(&fuzzPacket, 0, sizeof(fuzzPacket));
+            fuzzPacket.mPsdu = fuzzRecvBuff;
+            fuzzPacket.mChannel = g_RecvChannel;
+            fuzzPacket.mLength = (uint8_t)(otPlatRandomGet() % 127);
+
+            // Populate the length with random
+            for (uint8_t i = 0; i < fuzzPacket.mLength; i++)
+            {
+                fuzzRecvBuff[i] = (uint8_t)otPlatRandomGet();
+            }
+
+            // Clear the global flag
+            g_RecvChannel = 0;
+
+            countRecv++;
+            Log("recv %u bytes", fuzzPacket.mLength);
+
+            // Indicate the receive complete
+            otPlatRadioReceiveDone(aInstance, &fuzzPacket, kThreadError_None);
+        }
+    }
+
+    Log("%u packets received", countRecv);
+
+#if _WIN32
```

Comment:
On further inspection, using `delete` in `otInstanceFinalize()` is broken since we did not actually receive the memory from `new`.  We should replace that line with: `aInstance->~otInstance();`